subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/internal/BuildScriptBuilder.java

```diff
@@ -54,13 +61,28 @@ public BuildScriptBuilder fileComment(String comment) {
 
     /**
      * Adds a plugin to be applied
+     *
      * @param comment A description of why the plugin is required
      */
     public BuildScriptBuilder plugin(String comment, String pluginId) {
         plugins.put(pluginId, comment);
         return this;
     }
 
+    /**
+     * Adds the application plugin and sets the application main class name.
+     *
+     * @param mainClassName the application main class name
+     */
+    public BuildScriptBuilder applicationWithMainClassName(String mainClassName) {
```

Comment:
I'm not sure we should have domain specific methods on this builder, they really belong on the descriptor.